ui/framework/src/ui-framework/imodel-components/VisibilityTreeEventHandler.ts

```diff
@@ -143,8 +143,15 @@ export class VisibilityTreeEventHandler extends UnifiedSelectionTreeEventHandler
       );
   }
 
-  private async updateCheckboxes(affectedNodes?: string[]) {
-    const changes = await (affectedNodes ? this.collectAffectedNodesCheckboxInfos(affectedNodes) : this.collectAllNodesCheckboxInfos());
+  private async updateCheckboxes(affectedNodes?: string[], visibilityStatus?: Map<string, VisibilityStatus>) {
+    const checkBoxInfos = new Map<string, CheckBoxInfo>();
+    if (visibilityStatus) {
+      visibilityStatus.forEach((value, key) => {
+        checkBoxInfos.set(key, this.createCheckboxInfo(value));
+      });
+    }
+
+    const changes = visibilityStatus ? checkBoxInfos : await (affectedNodes ? this.collectAffectedNodesCheckboxInfos(affectedNodes) : this.collectAllNodesCheckboxInfos());

```

Comment:
I thinks this is not how `visibilityStatus` should be handled. With current implementation when `visibilityStatus` map contains one entry only that one node will be updated even if there are multiple `affectedNodes`.

I thinks `visibilityStatus` map should be handled in `getNodeCheckBoxInfo`. It's main reason is to avoid calling `getVisibilityStatus` if status for that node could be determined before emitting `onVisibilityChange` event. It should be possible to update all nodes and provide already determined status for couple of them.

Reply:
Sure, updated it